### PR TITLE
Revisit cache-remote option description

### LINKED ISSUE #34509: Revisit `cache-remote-*` option description
### Description

The description of these options mentions they replace items in the XML file. It is not true anymore since it is used to configure the Hot Rod client.

### PR DESCRIPTION
**Pull request by pruivo**
Closes #34509

### CODE REVIEW

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.approved.txt [hunk 1]
```diff
@@ -64,11 +64,7 @@ Cache:
 --cache-embedded-users-max-count <max-count>
                      The maximum number of entries that can be stored in-memory by the users cache.
 --cache-remote-host <hostname>
-                     The hostname of the remote server for the remote store configuration. It
-                       replaces the 'host' attribute of 'remote-server' tag of the configuration
-                       specified via XML file (see 'cache-config-file' option.). If the option is
-                       specified, 'cache-remote-username' and 'cache-remote-password' are required
-                       as well and the related configuration in XML file should not be present.
+                     The hostname of the external Infinispan cluster.
```
Comment (mhajas) [inline quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.approved.txt:67]: Should we add more details on when is this option used?

Comment (pruivo) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testStartDevHelp.approved.txt:67]: https://github.com/keycloak/keycloak/issues/34536 will add a message that this option is available when the feature multi-site is enabled.

**Comment by pruivo** [comment]
Rebased!

**Review by mhajas — APPROVED** [review]
Thank you @pruivo! LGTM.

**Pull request merged**